test/unit/acknowledgement/process-invalid.test.js

```diff
@@ -0,0 +1,242 @@
+const mockCommit = jest.fn()
+const mockRollback = jest.fn()
+const mockTransaction = {
+  commit: mockCommit,
+  rollback: mockRollback
+}
+
+jest.mock('../../../app/data', () => {
+  return {
+    sequelize:
+       {
+         transaction: jest.fn().mockImplementation(() => {
+           return { ...mockTransaction }
+         })
+       }
+  }
+})
+
+jest.mock('../../../app/config')
+const mockConfig = require('../../../app/config')
+
+jest.mock('../../../app/event')
+const mockEvent = require('../../../app/event')
+
+jest.mock('../../../app/acknowledgement/get-hold-category-name')
+const getHoldCategoryName = require('../../../app/acknowledgement/get-hold-category-name')
+
+jest.mock('../../../app/holds/get-hold-category-id')
+const getHoldCategoryId = require('../../../app/holds/get-hold-category-id')
+
+jest.mock('../../../app/reschedule/index')
+const holdAndReschedule = require('../../../app/reschedule/index')
+
+jest.mock('../../../app/reset/index')
+const { resetPaymentRequestById } = require('../../../app/reset/index')
+
+const mockFRN = require('../../mock-frn')
+const processInvalid = require('../../../app/acknowledgement/process-invalid')
+const { SFI } = require('../../../app/schemes')
+
+let mockAcknowledgement
+let mockAcknowledgementError
+let mockHoldCategoryName
+let mockHoldCategoryId
+let mockHoldReschedule
+let mockResetPaymentRequestById
+
+let schemeId
+let paymentRequestId
+
+describe('send acknowledgement error event', () => {
+  beforeEach(() => {
+    mockConfig.isAlerting = true
+
+    mockAcknowledgement = JSON.parse(JSON.stringify(require('../../mock-acknowledgement')))
+    mockAcknowledgementError = JSON.parse(JSON.stringify(require('../../mock-acknowledgement-error')))
+    mockHoldCategoryName = JSON.parse(JSON.stringify(require('../../constants/hold-categories-names')))
```

Comment:
`mockHoldCategoryName` is equal to `{DAX_REJECTION: 'DAX rejection', BANK_ACCOUNT_ANOMALY: 'Bank account anomaly'}`

what we probably want is it to be equal to one of these keys' value, i.e. we want this to match how our `mockAcknowledgementError` which has its message as `'Journal JN12345678 has been created Validation failed Line : 21.', looking at `getHoldCategoryName` this maps it to `'Dax rejection'`, therefore we've use this.

To do so we can use this syntax: `{DAX_REJECTION: mockHoldCategoryName } = JSON.parse(JSON.stringify(require('../../constants/hold-categories-names')))` (fancy way of renaming an object import)

Reply:
Would this syntax be replacing:mockHoldCategoryName = JSON.parse(JSON.stringify(require('../../constants/hold-categories-names')))

Reply:
Yup, would result in `mockHoldCategoryName` = `'DAX rejection'`